modules/cats/src/main/scala/io/funcqrs/cats/CommandHandlers.scala

```diff
@@ -0,0 +1,53 @@
+package io.funcqrs.cats
+
+import cats.Functor
+import cats.data.Validated.Invalid
+import cats.data.{NonEmptyList, ValidatedNel}
+import io.funcqrs.behavior.CommandToManyEvents
+import io.funcqrs.behavior.handlers.{CommandHandler, CommandHandlerInvoker}
+import io.funcqrs.interpreters.Identity
+
+import scala.collection.immutable
+import scala.language.higherKinds
+
+/**
+  * Users of `validated` should implement hierarchy of errors that has root of `ValidationError`.
+  */
+trait ValidationError
+
+/**
+  * @tparam VE - validation error type
+  * @tparam F2 - functor type
+  */
+trait ValidatedCommandHandlers[VE, F2[_]] {
```

Comment:
Hi @LGLO,
 
I initially didn't get your motivation for `F2`, but after going through `IfCompilesThenWorks.scala` I got the idea. That's neat idea! 

I'm wondering if we if we should extend it to the other `CommandHandlers` we have. 

We currently have support for:

-`Identity` (`just`)
- `Option` (`maybe`)
- `Try` (`attempt`)
- `Future` (`eventually`)

But we are not covering:
`Try[Option]` - (`attempt.maybe`) 
`Future[Option]` - (`eventually.maybe`) 

The consequence of extending this to the existing `CommandHandlers` is that we will need to add our own `Functor` type or add `Cats` as a main dependency.

Reply:
I see no problem depending on `Cats` for this.